apps/server/src/modules/board/repo/board-do.repo.spec.ts

```diff
@@ -193,18 +194,82 @@ describe(BoardDoRepo.name, () => {
 			return { board, column, card1: cards[0], card2: cards[1], card3: cards[2] };
 		};
 
-		it('should save the object', async () => {
-			const { board } = await setup();
+		it('should create new board nodes', async () => {
+			const cards = cardFactory.buildList(3);
 
-			await repo.save(board);
+			await repo.save(cards);
 			em.clear();
 
-			const result = await em.findOneOrFail(BoardNode, board.id);
+			const result = await em.find(CardNode, {});
+			expect(result.map((n) => n.id).sort()).toEqual(cards.map((c) => c.id).sort());
+		});
+
+		it('should update existing board nodes', async () => {
+			const node = cardNodeFactory.buildWithId({ title: 'before' });
+			await em.persistAndFlush(node);
+
+			const card = await repo.findByClassAndId(Card, node.id);
+			card.title = 'after';
+
+			await repo.save(card);
+			em.clear();
+
+			const result = await em.findOneOrFail(CardNode, node.id);
+			expect(result.title).toEqual('after');
+		});
+
+		it('should be able to do both - create and update', async () => {
+			const node1 = cardNodeFactory.buildWithId({ title: 'before' });
+			await em.persistAndFlush(node1);
+			em.clear();
+			const card1 = await repo.findByClassAndId(Card, node1.id);
+			card1.title = 'after';
+			const card2 = cardFactory.build({ title: 'created' });
+
+			await repo.save([card1, card2]);
+			em.clear();
 
-			expect(result.id).toEqual(board.id);
+			const result = await em.find(CardNode, {});
+			expect(result.map((n) => n.title).sort()).toEqual(['after', 'created']);
 		});
 
-		it('should persist order to positions', async () => {
+		describe('when parent is already persisted', () => {
+			it('should create child nodes', async () => {
+				const column = columnFactory.build();
+				await repo.save(column);
+
+				const cards = cardFactory.buildList(2);
+				cards.forEach((card) => column.addChild(card));
+				await repo.save(cards, column);
+
+				const result = await em.find(CardNode, {});
+				expect(result.map((n) => n.parentId)).toEqual([column.id, column.id]);
+			});
+		});
+
+		describe('when parent is newly built (not yet persisted)', () => {
+			it('should throw an error', async () => {
+				const card = cardFactory.build();
+				const column = columnFactory.build({ children: [card] });
+
+				await expect(repo.save(card, column)).rejects.toThrow();
+			});
+		});
+
+		describe('when objects have different parents', () => {
+			it('should throw an error', async () => {
+				const card1 = cardFactory.build();
+				const card2 = cardFactory.build();
+
+				const column1 = columnFactory.build({ children: [card1] });
+				const column2 = columnFactory.build({ children: [card2] });
+				await repo.save([column1, column2]);
```

Comment:
We have code style rules

https://docs.dbildungscloud.de/display/DBH/Code+Styles+08.02.23#CodeStyles08.02.23-setupfunctioninsteadofbeforeEach

Reply:
updated the test, see https://github.com/hpi-schul-cloud/schulcloud-server/pull/4093/commits/36a6f0c1ec9de7308c90e6fb050995ba51339db9